ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetDetailedObjects.java

```diff
@@ -62,55 +68,120 @@ public CliCommand init(final Arguments args) throws Exception {
-        suspectBulkObjects = FluentIterable.from(new LazyIterable<>(
+        final FluentIterable<DetailedS3Object> detailedObjects = FluentIterable.from(new LazyIterable<>(
                         new GetObjectsFullDetailsLoaderFactory(getClient(), this.bucketName, this.prefix, 100, 5, true)))
-                .filter(Predicates.notNull());
+                .filter(Predicates.and(getDatePredicate(), getSizePredicate(), getNamePredicate(), getOwnerPredicate()));
+
+        return new GetDetailedObjectsResult(detailedObjects);
+    }
 
-        if (filterPredicate != null) {
-            return new GetDetailedObjectsResult(suspectBulkObjects.filter(filterPredicate));
+    protected Predicate<DetailedS3Object> getSizePredicate() throws CommandException {
+        if (Guard.isMapNullOrEmpty(this.filterParams)) {
+            return Predicates.notNull();
         }
 
-        return new GetDetailedObjectsResult(suspectBulkObjects);
-    }
+        String larger = this.filterParams.get(LARGERTHAN);
+        String smaller = this.filterParams.get(SMALLERTHAN);
+        if (Guard.isStringNullOrEmpty(larger) && Guard.isStringNullOrEmpty(smaller)) {
+            return Predicates.notNull();
+        }
+        // if one is specified, use default value for other
+        final long largerthan = Guard.isStringNullOrEmpty(larger) ? 0L : Long.parseLong(larger);
+        final long smallerthan = Guard.isStringNullOrEmpty(smaller) ? Long.MAX_VALUE : Long.parseLong(smaller);
 
-    protected Predicate<DetailedS3Object> getPredicate() throws CommandException {
+        return new Predicate<DetailedS3Object>() {
+            @Override
+            public boolean apply(@Nullable final DetailedS3Object input) {
+                return input.getSize() > largerthan
+                        && input.getSize() < smallerthan;
+            }
+        };
+    }
 
+    protected Predicate<DetailedS3Object> getDatePredicate() throws CommandException {
         if (Guard.isMapNullOrEmpty(this.filterParams)) {
-            return null;
+            return Predicates.notNull();
         }
 
-        // else build a predicate from search-params
-        final Map<String, String> ranges = parseMeta();
-        final long largerthan = Long.parseLong(ranges.get(LARGERTHAN));
-        final long smallerthan = Long.parseLong(ranges.get(SMALLERTHAN));
-        final Date newerthan = new Date(Long.parseLong(ranges.get(NEWERTHAN)));
-        final Date olderthan = new Date(Long.parseLong(ranges.get(OLDERTHAN)));
+        String newer = this.filterParams.get(NEWERTHAN);
+        String older = this.filterParams.get(OLDERTHAN);
+        if (Guard.isStringNullOrEmpty(newer) && Guard.isStringNullOrEmpty(older)) {
+            return Predicates.notNull();
+        }
+        // if one is specified, default the other
+        final long newerThan = Guard.isStringNullOrEmpty(newer) ? 0L : new Date().getTime() - Utils.dateDiffToSeconds(newer) * 1000;
```

Comment:
What is the magic 1000 number doing?  It would be better to replace it with a constant.